### PR TITLE
WIP: Follow README style guide

### LINKED ISSUE #1139: A style guide for the README and the update log
Both the README and [the new update log](https://github.com/sindresorhus/refined-github/issues/1137) contain stylistic inconsistencies. They make the text harder to read and make Refined GitHub look less polished. That's why I wrote a little style guide:

### Formatting

* Keys should be written in lower case.
  * <kbd>shift</kbd> instead of <kbd>Shift</kbd>.
  * :arrow_right: *Used by GitHub.*
* Keys of key combos are only separated by a space.
  * <kbd>g</kbd> <kbd>t</kbd> instead of <kbd>g</kbd>+<kbd>t</kbd>.
  * :arrow_right: *Used by GitHub.*
* Mouse buttons ("clicks") are treated as keys.
  * <kbd>alt</kbd> <kbd>click</kbd> instead of <kbd>alt</kbd> click.
  * :arrow_right: *Just looks better. I guess.*
* Labels (of buttons, tabs, links, ...) are highlighted using backticks.
  * `Button` instead of 'Button' or "Button".
  * :arrow_right: *Already used in some places. Also more readable.*

### Wording

* Feature bullet points are sentences that start with an imaginary *Refined GitHub allows you to ...*.
  * *Toggle files* instead of *The ability to toggle files* or *Let's you toggle files*.

---

Anything to add or change? Otherwise, I'll open a PR to obey the rules in the README.

Should I also include (or link to?) the style guide in the contribution guide?

### PR DESCRIPTION
**Pull request by shroudedcode**
Obeys the style guide proposed in #1139.

*Closes #1139*

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Suggestions and pull requests are highly encouraged! Have a look at the [open is
 - Each JavaScript feature lives in its own file under [`source/features`](https://github.com/sindresorhus/refined-github/tree/master/source/features) and it's loaded on condition in [`source/content.js`](https://github.com/sindresorhus/refined-github/blob/master/source/content.js)
 - Some GitHub pages are loaded via AJAX/PJAX, so some features need to be in the special `ajaxedPagesHandler` function (see it as a custom "on DOM ready")
 - See what a _feature_ [looks like](https://github.com/sindresorhus/refined-github/blob/master/source/features/add-time-machine-links-to-comments.js) and [how it's loaded](https://github.com/sindresorhus/refined-github/blob/d7768508c3919558daa6b2ccc5a15aa73d081551/source/content.js#L176) (in this case it uses `observeEl` to automatically be run when new comments are loaded via AJAX)
+- If you're making changes to the README, try to match the style of the content that's already there and when in doubt, take a look at [our little style guide](https://github.com/sindresorhus/refined-github/issues/1139)
```
Comment (hkdobrev) [inline contributing.md]: I'll add a dot at the end of the sentence after the link here and on all other list items which don't have it here. It's just better as these are real sentences. I know this is not entirely related, but since this is about fixing markdown style..

File: readme.md [hunk 2]
```diff
@@ -90,11 +90,11 @@ GitHub Enterprise is also supported. More info in the options.
...
-- Copy canonical link to file when [the <kbd>y</kbd> hotkey](https://help.github.com/articles/getting-permanent-links-to-files/) is used
-- Supports indenting with the tab key in textareas like the comment box (<kbd>Shift</kbd> <kbd>Tab</kbd> for original behavior)
-- [Uses the pull request title as commit title when merging with 'Squash and merge'](https://github.com/sindresorhus/refined-github/issues/276)
+- [Makes it possible to automatically merge a PR after its build has passed](https://user-images.githubusercontent.com/1402241/35192861-3f4a1bf6-fecc-11e7-8b9f-35ee019c6cdf.gif)
+- Allows you to toggle all [outdated PR comments](https://user-images.githubusercontent.com/25818354/33240033-3e271588-d2af-11e7-93af-13b6e325f65d.gif) or [PR/Commit files](https://user-images.githubusercontent.com/1402241/35192652-6f79dc42-fec9-11e7-89ad-2b4a9c5f4f52.gif) with <kbd>alt</kbd> <kbd>click</kbd>
+- Copies canonical link to file when [the <kbd>y</kbd> hotkey](https://help.github.com/articles/getting-permanent-links-to-files/) is used
+- Supports indenting with the tab key in textareas like the comment box (<kbd>shift</kbd> <kbd>tab</kbd> for original behavior)
```
Comment (hkdobrev) [inline readme.md]: The first "tab" word here should be <kbd>tab</kbd>.

**Comment by hkdobrev** [comment]
This could probably be tackled in a separate PR, but I'd recommend imperative language for describing added features:

"Copy canonical link to file" vs "Copies canonical link to file" across all features. This follows popular commit convention by Linus Torvalds. The reasoninng is that you instruct Git to apply that change if you pick this commit. In the same way, you instruct the extension if you have that feature. It's similar to changelog formats based on commit messages following that convention. I guess I'm probably used to that, but I still like the brevity and clarity of the imperative form.

File: contributing.md (hunk 1 above)
Comment (bfred-it) [reply contributing.md]: None of the above items are either, they'll have to be updated

**Comment by bfred-it** [comment]
@hkdobrev sounds good.

I'd add _dropping lets you_/_allows_. Examples:

```diff
- Allows you to toggle all outdated PR comments or PR/Commit files with <kbd>alt</kbd> <kbd>click</kbd>
+ Toggle all outdated PR comments or PR/Commit files with <kbd>alt</kbd> <kbd>click</kbd>

- Makes it possible to automatically merge a PR after its build has passed
+ Automatically merge a PR after its build has passed (checkbox)

- Supports indenting with the tab key in textareas like the comment box (<kbd>shift</kbd> <kbd>tab</kbd> for original behavior)
+ Indent with the <kbd>tab</kbd> key in textareas like the comment box (<kbd>shift</kbd> <kbd>tab</kbd> for original behavior)
```

**Comment by shroudedcode** [comment]
@hkdobrev @bfred-it There's a difference between Refined GitHub doing something by default (decluttering the UI) and it allowing the user to do something (copying a link).

Using the same style for both cases might make users think Refined GitHub does things by default that actually have to be explicitly triggered/enabled by them.

**Comment by bfred-it** [comment]
For reference, that’s also how Apple talks about what their software *lets you do*: 
6d235b33-083f-4ab2-bc56-73d9223ec8a5

File: contributing.md (hunk 1 above)
Comment (hkdobrev) [reply contributing.md]: The first two items in the list have full stops at the end.

**Comment by shroudedcode** [comment]
> For reference, that’s also how Apple talks about what their software lets you do:

I like that style, but it requires adding the features' *uses* to the current descriptions that just say *what they do*:

* Adds a `Trending` link to the global navbar. *(<kbd>g</kbd> <kbd>t</kbd> hotkey)*
  * Access the trending repositories with a link in the global navbar or the <kbd>g</kbd> <kbd>t</kbd> hotkey.
* [Adds <kbd>↑</kbd> shortcut for editing your last comment](https://github.com/sindresorhus/refined-github/pull/961)
  * [Quickly edit your last comment using the <kbd>↑</kbd> shortcut.](https://github.com/sindresorhus/refined-github/pull/961)
* [Moves destructive buttons in commenting forms away from the primary button](https://user-images.githubusercontent.com/170270/27985285-a5464fb2-63e8-11e7-8c01-f33bb7ad78ba.png)
  * [No longer click destructive buttons on accident as they're moved away from the primary button.](https://user-images.githubusercontent.com/170270/27985285-a5464fb2-63e8-11e7-8c01-f33bb7ad78ba.png)

What do you think about rewording the features in this way?

**Comment by bfred-it** [comment]
Love it! The third one is a bit awkward, maybe it could be made passive:

- Destructive buttons are moved away from the primary button

File: contributing.md [hunk 3]
```diff
@@ -42,17 +43,17 @@ Once built, load it in the browser of your choice:
...
 			</ol>
 		</td>
 		<td width="50%">
 			<ol>
-				<li>Open <code>about:debugging#addons</code>
-				<li>Click on the <strong>Load Temporary Add-on</strong> button
-				<li>Select the file <code>refined-github/distribution/manifest.json</code>
+				<li>Open <code>about:debugging#addons</code>.
+				<li>Click on the <strong>Load Temporary Add-on</strong> button.
+				<li>Select the file <code>refined-github/distribution/manifest.json</code>.
```
Comment (bfred-it) [inline contributing.md]: These should end with `;` except the last one, which ends with `.`

**Comment by ramlmn** [comment]
Random ping: @shroudedcode resolve this before we have any merge conflicts later!

**Comment by shroudedcode** [comment]
@ramlmn I'll fix the merge conflict on the weekend. I'm doing an internship this week and can barely focus on doing any kind of open source work in the evening. Also, stop pinging people about their pull requests unless the pull request is really old **and** they have been active within the last week (a sign that they might have actually forgotten about it). Often they have reasons for "abandoning" them for a while.

**Comment by shroudedcode** [comment]
The pull request is by the way ready for review again. @bfred-it @hkdobrev

File: package.json [hunk 4]
```diff
@@ -120,6 +120,9 @@
 			"rule-empty-line-before": null,
 			"at-rule-empty-line-before": null,
 			"selector-max-compound-selectors": null
-		}
+		},
+		"ignoreFiles": [
+			"**/*.md"
+		]
```
Comment (bfred-it) [inline package.json]: This doesn't seem to change the linter's output

File: readme.md [hunk 5]
```diff
@@ -73,10 +73,10 @@ GitHub Enterprise is also supported. More info in the options.
 <table>
 	<tr>
 		<th width="50%">
-			Linkifies issue/PR references in code, comments and titles
+			Click on references to issues and pull requests
```
Comment (bfred-it) [inline readme.md]: Maybe "Clickable references"

File: readme.md [hunk 6]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
-- Supports indenting with the tab key in textareas like the comment box (<kbd>Shift</kbd> <kbd>Tab</kbd> for original behavior)
-- [Uses the pull request title as commit title when merging with 'Squash and merge'](https://github.com/sindresorhus/refined-github/issues/276)
-- [Automatically embeds linked gists in comments](https://user-images.githubusercontent.com/6978877/33911900-c62ee968-df8b-11e7-8685-506ffafc60b4.PNG)
-- [Adds scoped search on each users profile page](https://user-images.githubusercontent.com/1402241/35185441-24ad4b1e-fe37-11e7-9e1b-0dc09fc1ada2.png)
+- Toggle all [outdated PR comments](https://user-images.githubusercontent.com/25818354/33240033-3e271588-d2af-11e7-93af-13b6e325f65d.gif) or [PR/commit files](https://user-images.githubusercontent.com/1402241/35192652-6f79dc42-fec9-11e7-89ad-2b4a9c5f4f52.gif) with <kbd>alt</kbd> <kbd>click</kbd>.
+- Copy the canonical link of a file with [the <kbd>y</kbd> hotkey](https://help.github.com/articles/getting-permanent-links-to-files/).
+- Indent text with <kbd>tab</kbd> in textareas like the comment box (<kbd>shift</kbd> <kbd>tab</kbd> for original behavior).
```
Comment (bfred-it) [inline readme.md]: Maybe `textarea`s or just "comment fields" (I don't know of any other textareas on this site)

File: readme.md [hunk 7]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Use the pull request title as the commit title when merging with `Squash and merge`](https://github.com/sindresorhus/refined-github/issues/276).
+- [Directly view linked gists inside comments without visiting the link.](https://user-images.githubusercontent.com/6978877/33911900-c62ee968-df8b-11e7-8685-506ffafc60b4.PNG)
```
Comment (bfred-it) [inline readme.md]: I think "embed" is clear enough to developers

File: readme.md [hunk 8]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Search a user's content using the user-scoped search on their profile page.](https://user-images.githubusercontent.com/1402241/35185441-24ad4b1e-fe37-11e7-9e1b-0dc09fc1ada2.png)
```
Comment (bfred-it) [inline readme.md]: Maybe

> Searches on user profiles are limited to the user.

File: readme.md [hunk 9]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
-- [Adds a link in a user's profile to their public gists](https://user-images.githubusercontent.com/11544418/34268306-1c974fd2-e678-11e7-9e82-861dfe7add22.png)
-- [Adds a 'Jump to bottom' link to discussion pages](https://user-images.githubusercontent.com/4331946/34298950-93876584-e720-11e7-898a-96f85e31aefe.png)
-- [Adds a 'Your repositories' link to user profile dropdown](https://user-images.githubusercontent.com/4201088/34920280-479e2454-f996-11e7-8e24-ad69793b9d9b.png)
-- [Adds a 'Download folder' button to repos](https://user-images.githubusercontent.com/1402241/35044451-fd3e2326-fbc2-11e7-82e1-61ec7bee612b.png)
-- [Adds a `Toggle files` button to repos](https://user-images.githubusercontent.com/1402241/35480123-68b9af1a-043a-11e8-8934-3ead3cff8328.gif)
-- [Adds links to the default branch and to the latest semver release](https://user-images.githubusercontent.com/1402241/36961980-389bd2b0-2080-11e8-8bff-bb8af52e7128.png)
+- [Click on the branch references in pull request to navigate to the branches.](https://github.com/sindresorhus/refined-github/issues/1)
```
Comment (bfred-it) [inline readme.md]: Like above, this is a bit wordy and it might just be "Clickable branch references"

File: readme.md [hunk 10]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Quickly edit a repository's README with the quick edit button.](https://user-images.githubusercontent.com/170270/27501200-31a1fa20-586c-11e7-9a3f-ce270014bf0a.png)
```
Comment (bfred-it) [inline readme.md]: > Quickly edit a repository's README from the repository root.

File: readme.md [hunk 11]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Quickly delete a forked repo after its pull request has been merged.](https://cloud.githubusercontent.com/assets/170270/13520281/b2c9335c-e211-11e5-9e36-b0f325166356.png)
+- [Flick through your notifications by opening them in seperate tabs.](https://user-images.githubusercontent.com/1402241/31700005-1b3be428-b38c-11e7-90a6-8f572968993b.png)
```
Comment (bfred-it) [inline readme.md]: "Flick" might sound like an actual gesture. Maybe just:

> Open all your notifications at once.

File: readme.md [hunk 12]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [View file diffs without whitespace changes.](https://cloud.githubusercontent.com/assets/170270/17603894/7b71a166-6013-11e6-81b8-22950ab8bce3.png) *(<kbd>d</kbd> <kbd>w</kbd> hotkey)*
```
Comment (bfred-it) [inline readme.md]: Maybe

> Easily toggle whitespace visibility in diffs.

File: readme.md [hunk 13]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Copy a file's content using the `Copy` button.](https://cloud.githubusercontent.com/assets/170270/14453865/8abeaefe-00c1-11e6-8718-9406cee1dc0d.png)
+- [Copy a gist file's content using the `Copy`.](https://cloud.githubusercontent.com/assets/170270/21074840/5dc37578-bf03-11e6-9fd9-501d73edef87.png)
```
Comment (bfred-it) [inline readme.md]: Maybe we can skip this altogether. The first one is good enough for both.

File: readme.md [hunk 14]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Copy a file's path using the `Copy path` button.](https://cloud.githubusercontent.com/assets/4201088/26023064/18c9c77c-37d2-11e7-8926-b0a05a2706ae.png)
```
Comment (bfred-it) [inline readme.md]: This only appears in PRs, so it needs to mention PRs.

File: readme.md [hunk 15]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Quickly access a commit's patch and  diff using links.](https://cloud.githubusercontent.com/assets/737065/13605562/22faa79e-e516-11e5-80db-2da6aa7965ac.png)
```
Comment (bfred-it) [inline readme.md]: There's an extra space before "diff".

File: readme.md [hunk 16]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [View a repository at the time of a comment.](https://user-images.githubusercontent.com/1402241/32310022-7fef6174-bf5d-11e7-960f-5041a8f073ac.png)
```
Comment (bfred-it) [inline readme.md]: Let's match GitHub's own wording:

> Browse a repository

Seen next to each commit: https://github.com/caitp/node-bluebird/commits/master

File: readme.md [hunk 17]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Quickly access a user's public gists from their profile.](https://user-images.githubusercontent.com/11544418/34268306-1c974fd2-e678-11e7-9e82-861dfe7add22.png)
```
Comment (bfred-it) [inline readme.md]: No "Quickly"

Also maybe just "Visit"

File: readme.md [hunk 18]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Jump to the bottom of a discussion page using the `Jump to bottom` link.](https://user-images.githubusercontent.com/4331946/34298950-93876584-e720-11e7-898a-96f85e31aefe.png)
```
Comment (bfred-it) [inline readme.md]: > Jump to the bottom of a discussion page _with a click_

File: readme.md [hunk 19]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Quickly access `Your repositories` from the profile dropdown.](https://user-images.githubusercontent.com/4201088/34920280-479e2454-f996-11e7-8e24-ad69793b9d9b.png)
```
Comment (bfred-it) [inline readme.md]: ~~Quickly~~

File: readme.md [hunk 20]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Download entire folders from repositories using the `Download folder` button.](https://user-images.githubusercontent.com/1402241/35044451-fd3e2326-fbc2-11e7-82e1-61ec7bee612b.png)
```
Comment (bfred-it) [inline readme.md]: Let's add:

> (it uses [download-directory.github.io](https://download-directory.github.io))

File: readme.md [hunk 21]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- [Collapse or expand a repository's files using the :arrow_down_small: button.](https://user-images.githubusercontent.com/1402241/35480123-68b9af1a-043a-11e8-8934-3ead3cff8328.gif)
+- [Quickly navigate to a repository's default branch or latest version tag using links.](https://user-images.githubusercontent.com/1402241/36961980-389bd2b0-2080-11e8-8bff-bb8af52e7128.png)
```
Comment (bfred-it) [inline readme.md]: > Quickly visit a repository's default branch and latest version tag

File: readme.md [hunk 22]
```diff
@@ -94,89 +94,92 @@ GitHub Enterprise is also supported. More info in the options.
...
+- The news feeds automagically expands when you scroll down.
+- The reaction popover is shown on hover instead of click.
+- The default sort order of milestones is changed to `Closest due date`.
+- The default sort order of issues and pull requests is changed to `Recently updated`.
+- [The `Expand diff` button is widened to make interaction easier.](https://user-images.githubusercontent.com/6978877/34470024-eee4f43e-ef20-11e7-9036-65094bd58960.PNG)
+- [Modals are automatically closed when they’re no longer visible.](https://user-images.githubusercontent.com/1402241/37022353-531c676e-2155-11e8-96cc-80d934bb22e0.gif)
+- [Quickly see which comment a permalink links to.](https://user-images.githubusercontent.com/1402241/37349492-226bd37a-2709-11e8-8087-d9686b330240.png)
+- [Inactive deployments in PR timelines are hidden.](https://github.com/sindresorhus/refined-github/issues/1144)
+
+And [many](source/content.css) [more...](source/content.js)
```
Comment (bfred-it) [inline readme.md]:
```md
And [many more…](source/content.css)
```

`content.js` no longer contains any features

**Comment by bfred-it** [comment]
Sorry! Lots of notes. I should have just made the changes where I was sure

File: package.json (hunk 4 above)
Comment (shroudedcode) [reply package.json]: I don't know why but without this option [the VS Code `stylelint` extension](https://marketplace.visualstudio.com/items?itemName=shinnn.stylelint) displays all sorts of errors in the `readme.md` file as if it was a stylesheet.

**Comment by shroudedcode** [comment]
Regarding the things I haven't changed yet: I'm trying to use the same style for all features: A sentence starting with an imaginary *Refined GitHub allows you to …* Having features that use a different style in between the others looks weird and is harder to read. I agree that some of the features still sound kind of awkward. If you have any suggestions on how to improve them while adhering to the style, let me know.

**Comment by bfred-it** [comment]
I don’t think we can stick to just that. It’s preferrable to keep sentences short and readable instead, given that there’s a lot of them. 

- People have fled

Is preferrable to

- The boats have allowed people to flee

File: readme.md [hunk 23]
```diff
@@ -99,7 +99,7 @@ GitHub Enterprise is also supported. More info in the options.
 - Copy the canonical link of a file with [the <kbd>y</kbd> hotkey](https://help.github.com/articles/getting-permanent-links-to-files/).
 - Indent text with <kbd>tab</kbd> inside comment fields (<kbd>shift</kbd> <kbd>tab</kbd> for original behavior).
 - [Use the pull request title as the commit title when merging with `Squash and merge`](https://github.com/sindresorhus/refined-github/issues/276).
-- [Directly view linked gists inside comments.](https://user-images.githubusercontent.com/6978877/33911900-c62ee968-df8b-11e7-8685-506ffafc60b4.PNG)
+- [Directly view linked gists inside comments.](https://user-images.githubusercontent.com/6978877/33911900-c62ee968-df8b-11e7-8685-506ffafc60b4.png)
```
Comment (bfred-it) [inline readme.md]: This new URL doesn't work. If you want to change case you'll have to re-upload it

Comment (shroudedcode) [reply readme.md]: Woah, that's weird. Did GitHub generate `.PNG` URLs in the past? I'll just remove the commit.

Comment (bfred-it) [reply readme.md]: No, GitHub just preserves the extension as-is, so if you upload `FILE.PNG` you'll get `2cb44.PNG`

**Comment by bfred-it** [comment]
Alright let's merge this asap, at least partially, @shroudedcode doesn't have to deal with insane merge conflicts

**Comment by bfred-it** [comment]
@shroudedcode thanks for this PR!

I just had to make many of them _passive_ because this was just too awkward:

> Quickly determine what file you're looking at with the sticky file headers on pull requests.

and can just be:

> File headers are always visible when scrolling.
